natchez-fs2/src/test/scala/com/ovoenergy/effect/natchez/AllocatedSpanTest.scala

```diff
@@ -0,0 +1,132 @@
+package com.ovoenergy.effect.natchez
+
+import java.util.UUID
+
+import cats.effect.{Clock, ContextShift, ExitCase, IO, Timer}
+import fs2.Stream
+import org.scalatest.Inspectors
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.concurrent.ExecutionContext
+
+class AllocatedSpanTest extends AnyWordSpec with Matchers {
+
+  implicit val cs: ContextShift[IO] =
+    IO.contextShift(ExecutionContext.global)
+
+  implicit val timer: Timer[IO] =
+    IO.timer(ExecutionContext.global)
+
+  implicit val clock: Clock[IO] =
+    Clock.extractFromTimer(timer)
+
+  val testEp: IO[TestEntryPoint[IO]] =
+    TestEntryPoint.apply[IO]
+
+
+  "AllocatedSpan.create" should {
+
+    "Submit the span even if a pre-submit task fails" in {
+      testEp
+        .flatMap { ep =>
+          val stream: IO[Unit] =
+            Stream
+              .emit("foo")
+              .covary[IO]
+              .through(AllocatedSpan.create(maxOpen = 10)(_ => ep.root("root")))
+              .evalTap(_.span.addSubmitTask(IO.raiseError(new Exception)).submit)
+              .compile
+              .drain
+
+          for {
+            _ <- stream.attempt
+            spans <- ep.spans
+          } yield {
+            spans.length shouldBe 1
+            spans.maxBy(_.completed).name shouldBe "root"
+            Inspectors.forAll(spans)(_.exitCase shouldBe ExitCase.Completed)
+          }
+        }
+        .unsafeRunSync()
+    }
+
+
+    "Hold the parent span open until it is explicitly submitted" in {
+      testEp
+        .flatMap { ep =>
+          val stream: IO[Unit] =
+            Stream
+              .emit("foo")
+              .covary[IO]
+              .through(AllocatedSpan.create(maxOpen = 10)(_ => ep.root("root")))
+              .evalTap(_.span.span("foo").use(_ => IO.unit))
```

Comment:
Just to clarify, if you wanted to create another sibling span you'd just call evalTap again and use the root span to create another child `_.span.span("bar").use...`?

Reply:
yep indeed :+1: